src/Streaming/Process.hs

```diff
@@ -180,17 +180,18 @@ uncons  stream = loop stream
 >  splitAt :: Control.Monad m => Int -> Stream (Of a) m r #-> Stream (Of a) m (Stream (Of a) m r)
 
 -}
-splitAt :: (HasCallStack, Control.Monad m, Control.Functor f) =>
+splitAt :: forall f m r. (Control.Monad m, Control.Functor f) =>
   Int -> Stream f m r #-> Stream f m (Stream f m r)
-splitAt n stream = Prelude.compare n 0 & \case
-  LT -> Prelude.error "splitAt called with negative integer" $ stream
```

Comment:
I could have an inner non-bound checking function that's identical and is optimized but this might (it seems to me) interplay badly with other things so I just avoided it.

Reply:
I'm still not seeing it. Why would the current version be more or less optimised than the previous one?